src/transformers/configuration_utils.py

```diff
@@ -306,7 +306,7 @@ def __init__(self, **kwargs):
         if self.id2label is not None:
             num_labels = kwargs.pop("num_labels", None)
             if num_labels is not None and len(self.id2label) != num_labels:
-                logger.warn(
+                logger.warning(
```

Comment:
I think that's our internal logger here no - not Python's logger. So not sure about this change @LysandreJik

Reply:
The logging module imported in this script is indeed our internal logging module that lives under the `utils` directory. We see here that the logger is instantiated by invoking the `get_logger` function from this module:

https://github.com/huggingface/transformers/blob/72728be3dbca26c70dddc8b724eb2c8d901e97dc/src/transformers/configuration_utils.py#L49

If we inspect the internal logging module in `utils`, we see that it first imports the Python logging module:
https://github.com/huggingface/transformers/blob/72728be3dbca26c70dddc8b724eb2c8d901e97dc/src/transformers/utils/logging.py#L17
The `get_logger` function simply returns the `getLogger` function from this Python logging module:
https://github.com/huggingface/transformers/blob/72728be3dbca26c70dddc8b724eb2c8d901e97dc/src/transformers/utils/logging.py#L113-L124

Hence, our internal logger is implicitly derived from its parent Python logger, and is simply a wrapper for this module. Consequently, the advice regarding the use of `warning` in place of `warn` should still hold.

Reply:
Yes, it's a good change!